File: app/src/main/java/de/bikebean/app/db/settings/settings/location_settings/Location.java
package de.bikebean.app.db.settings.settings.location_settings;

import androidx.annotation.NonNull;

import de.bikebean.app.db.settings.settings.LocationSetting;
import de.bikebean.app.db.sms.Sms;
import de.bikebean.app.db.state.State;

public class Location extends LocationSetting {

    private final static @NonNull State.KEY key = State.KEY.LOCATION;

    public Location(double location, final @NonNull Sms sms) {
        super(location, sms, key);
    }

    public Location(final @NonNull Sms sms) {
        super(sms, key);
    }

    public Location() {
        super(key);
    }
}

File: app/src/main/java/de/bikebean/app/db/settings/settings/add_to_list_settings/location_settings/Location.kt
package de.bikebean.app.db.settings.settings.add_to_list_settings.location_settings

import de.bikebean.app.db.settings.settings.add_to_list_settings.LocationSetting
import de.bikebean.app.db.settings.settings.add_to_list_settings.WappState
import de.bikebean.app.db.state.State.KEY

class Location : LocationSetting {

    constructor(
            location: Double,
            wappState: WappState
    ) : super(location, wappState, key)

    constructor(wappState: WappState) : super(wappState.sms, key)

    constructor() : super(key)

    companion object {
        private val key = KEY.LOCATION
    }

}